### PR TITLE
Config API

### PR DESCRIPTION
**Pull request by dzwdz**
this is not quite ready yet as it'll be refactored into one ConfigManager class, but @YTG1234 told me to make a draft pr so here it is

### CODE REVIEW

**Comment by YTG1234** [comment]
Will the `toJson` method be moved into the `ConfigManager`?

**Comment by dzwdz** [comment]
yes, from/toJson and getConfigurableFields will all be moved to ConfigManager, along with the functions handling reading/writing the actual config file

File: build.gradle
Comment (YTG1234) [inline build.gradle]: We kinda already have a repositories block at the top of the file

Comment (dzwdz) [reply build.gradle]: 👀

**Comment by YTG1234** [comment]
so... I guess using reflection in fromJson as well! Great!

I don't hate reflection or something, I just think it doesn't look good in code

File: build.gradle
Comment (YTG1234) [reply build.gradle]: Kara why did you mark this as resolved

**Review by YTG1234 — APPROVED** [review]
Pretty solid config system, very cool.

**Comment by YTG1234** [comment]
@dzwdz What's left to mark this as ready for review?

**Comment by YTG1234** [comment]
this should be an external API lol

**Review by YTG1234 — APPROVED** [review]
Approving again because commit

**Comment by dzwdz** [comment]
don't merge yet, i'll move the exception handling out of ConfigManager

**Review by YTG1234 — APPROVED** [review]
Approving because more commits

**Comment by dzwdz** [comment]
this is the probably the final version

**Review by YTG1234 — APPROVED** [review]
👍

**Review by namebee — APPROVED** [review]

**Comment by dzwdz** [comment]
was that commit really necessary? you could've done it after the merge

**Review by YTG1234 — APPROVED** [review]
yes

**Comment by YTG1234** [comment]
Didn't wait for review from @hYdos ? Fine I guess

**Pull request merged**
